airflow-core/src/airflow/models/dag.py

```diff
@@ -1966,6 +1966,10 @@ class DagModel(Base):
         cascade="all, delete, delete-orphan",
     )
     schedule_assets = association_proxy("schedule_asset_references", "asset")
+    task_inlet_asset_references = relationship(
+        "TaskInletAssetReference",
+        cascade="all, delete, delete-orphan",
+    )
```

Comment:
Nice, looking forward to it so we can have a 'task_level' representation of Asset inlets for the graph view. Resolved aliases do not use that if I'm not mistaken and we will still need to do an extra lookup, but that fine